File: 0x08-python-more_classes/4-rectangle.py
#!/usr/bin/python3
"""class rectangle created"""


class Rectangle:
    """empty class rectangle defined"""
    def __init__(self, width=0, height=0):
        """defined method"""
        self.height = height
        self.width = width
        """Private instance attribute"""

    @property
    def width(self):
        return self.__width

    @width.setter
    def width(self, value):
        self.__width = value
        if not isinstance(value, int):
            raise TypeError("width must be an integer")
        if value < 0:
            raise ValueError("width must be >= 0")

    @property
    def height(self):
        return self.__height

    @height.setter
    def height(self, value):
        self.__height = value
        if not isinstance(value, int):
            raise TypeError("height must be an integer")
        if value < 0:
            raise ValueError("height must be >= 0")

    def area(self):
        return (self.__width * self.__height)

    def perimeter(self):
        if self.__width == 0 or self.__height == 0:
            return 0
        return (self.__width * 2 + self.__height * 2)

    def __str__(self) -> str:
        var = ""
        if self.__width == 0 or self.__height == 0:
            return ""
        for i in range(self.__height):
            for j in range(self.__width):
                var += "#"
            if (self.__height - 1) != i:
                var += "\n"
        return var

    def __repr__(self):
        return ("Rectangle ({:d}, {:d})".format(self.__width, self.__height))

File: 0x13-javascript_objects_scopes_closures/4-rectangle.js
#!/usr/bin/node

module.exports = class Rectangle {
	constructor (w, h) {
	  if (w <= 0 || h <= 0) {
		return 0;
	  } else if (!w || !h) {
		return 0;
	  }
	  this.width = w;
	  this.height = h;
	}
  
	print () {
	  let size;
	  let j;
  
	  for (size = 0; size < this.height; size++) {
		let row = '';
		for (j = 0; j < this.width; j++) {
		  row += 'X';
		}
		console.log(row);
	  }
	}

	rotate () {
	  let temp;

	  temp = this.height;
	  this.height = this.width;
	  this.width = temp;
	}

	double () {
		this.width = this.width * 2;
		this.height = this.height * 2;
	}
  };
  